engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/batch/NadelBatchHydrationByObjectId.kt

```diff
@@ -69,72 +123,42 @@ internal object NadelBatchHydrationByObjectId {
     private fun getHydrateInstructionsForNodeMatchingObjectId(
         executionBlueprint: NadelOverallExecutionBlueprint,
         state: NadelBatchHydrationTransform.State,
-        parentNode: JsonNode,
-        parentNodeIdentifierNodes: List<JsonNode>,
-        resultNodesByObjectId: Map<Any?, JsonMap>,
+        sourceNode: JsonNode,
+        sourceIds: List<List<Any?>>,
+        resultNodesByObjectId: Map<List<Any?>, JsonMap>,
     ): NadelResultInstruction {
         val hydratedFieldDef = NadelTransformUtil.getOverallFieldDef(
             overallField = state.hydratedField,
-            parentNode = parentNode,
+            parentNode = sourceNode,
             service = state.hydratedFieldService,
             executionBlueprint = executionBlueprint,
             aliasHelper = state.aliasHelper,
         ) ?: error("Unable to find field definition for ${state.hydratedField.queryPath}")
 
         val newValue: Any? = if (hydratedFieldDef.type.unwrapNonNull().isList) {
             // Set to null if there were no identifier nodes
-            if (parentNodeIdentifierNodes.isNotEmpty() && parentNodeIdentifierNodes.all { it.value == null }) {
+            if (sourceIds.isNotEmpty() && sourceIds.all { it.all { it == null } }) {
                 null
             } else {
-                parentNodeIdentifierNodes
+                sourceIds
                     .asSequence()
-                    .map { it.value }
-                    .flatten(recursively = true)
-                    .filterNotNull()
+                    .map { it }
+                    // .flatten(recursively = true)
+                    // .filterNotNull()
                     .map { id ->
                         resultNodesByObjectId[id]
                     }
                     .toList()
             }
         } else {
-            parentNodeIdentifierNodes.emptyOrSingle()?.let { node ->
-                resultNodesByObjectId[node.value]
+            sourceIds.emptyOrSingle()?.let { sourceId ->
+                resultNodesByObjectId[sourceId]
             }
         }
 
         return NadelResultInstruction.Set(
-            subjectPath = parentNode.resultPath + state.hydratedField.resultKey,
+            subjectPath = sourceNode.resultPath + state.hydratedField.resultKey,
             newValue = newValue,
         )
     }
-
-    /**
-     * For the following example
-     *
-     * ```
-     * type Issue {
-     *   details: IssueDetails
-     *   owner: User @hydrated(from: ["issueOwner"], args: [
-     *     {name: "issueId" valueFromField: ["details", "authorId"]}
-     *   ])
-     * }
-     *
-     * type IssueDetails {
-     *   authorId: ID!
-     * }
-     * ```
-     *
-     * We are getting the path `["details", "authorId"]`
-     */
-    private fun getPathToObjectIdentifierOnHydrationParentNode(
```

Comment:
This is now stored in `NadelBatchHydrationMatchStrategy.MatchObjectIdentifier.sourceId`